Source/Orts.Simulation/Common/Scripting/TrainControlSystem.cs

```diff
@@ -667,6 +673,24 @@ public SignalFeatures(string mainHeadSignalTypeName, Aspect aspect, float distan
         }
     }
 
+    public struct SpeedPostFeatures
+    {
+        public readonly string SpeedPostTypeName;
```

Comment:
I suggest using `SpeedPostShapeName` as the property name here. It would be more descriptive.

Reply:
If, one day, we add the possibility to add new signals or speed posts from packages (which seems to be something that the virtual filesystem concept can manage to do), I think we would have config files (in JSON for example) for each new signal or each new speed post.
In that context, I think the term "speed post type name" is more generic and can be reused later for speed posts with a JSON config file.

Reply:
What gets me is that you have an equivalent SignalFeatures property, MainHeadSignalTypeName, that communicates the MSTS signal type: "NORMAL", "DISTANCE", "SPEED", etc. To a user who doesn't understand your intentions for SpeedPostTypeName, it looks like a completely different piece of data that, for some reason, shares the same name.

To me, the speed post **type name** would be something more like the **TrainObjectType.SpeedItemType** enum, which can be one of "Standard", "TempRestrictedStart", and "TempRestrictedResume". If you want access to the shape name, then I still think it would be better to use an unambiguous "SpeedPostShapeName" property. (Eventually, when we get new file formats and editor support (ugh), we can introduce custom identifiers.)

Reply:
> What gets me is that you have an equivalent SignalFeatures property, MainHeadSignalTypeName, that communicates the MSTS signal type: "NORMAL", "DISTANCE", "SPEED", etc. To a user who doesn't understand your intentions for SpeedPostTypeName, it looks like a completely different piece of data that, for some reason, shares the same name.

No, it contains the signal type name of the SIGCFG file, not the signal function type:
![image](https://user-images.githubusercontent.com/375720/123374299-2d2a9a00-d587-11eb-9f5e-9c94c3a15a62.png)

> To me, the speed post **type name** would be something more like the **TrainObjectType.SpeedItemType** enum, which can be one of "Standard", "TempRestrictedStart", and "TempRestrictedResume". If you want access to the shape name, then I still think it would be better to use an unambiguous "SpeedPostShapeName" property. (Eventually, when we get new file formats and editor support (ugh), we can introduce custom identifiers.)

That would be the function type of the speed post.

Reply:
Huh! I totally misread that code.

Okay, then no objections from me. Thanks for clarifying this.